src/codec.rs

```diff
@@ -1227,6 +1230,29 @@ impl Decode for () {
 	}
 }
 
+#[cfg(feature = "bit-vec")]
+impl Encode for BitVec {
+	fn encode_to<W: Output>(&self, dest: &mut W) {
+		let len = self.len();
+		assert!(len <= u32::max_value() as usize, "Attempted to serialize a collection with too many elements.");
+		Compact(len as u32).encode_to(dest);
```

Comment:
Why do we encode the length twice?

Reply:
I agree that it is somewhat ugly, but these are different lengths. The first one is a number of bits in `BitVec` and the second one is the length of a slice in elements (bytes):
```rust
let vec = bitvec![0, 0];
assert_eq!(2, vec.len());
assert_eq!(1, vec.as_slice().len());
```

Reply:
Ahh okay, but as the length of slices is computable by the number of bits, you should just do 
`dest.write(self.as_slice())` to write without the second length.